src/main/java/com/google/netpcapanalysis/dao/MaliciousIPDaoImpl.java

```diff
@@ -0,0 +1,64 @@
+package com.google.netpcapanalysis.dao;
+import com.google.netpcapanalysis.dao.PCAPDaoImpl;
+import com.google.netpcapanalysis.interfaces.dao.PCAPDao;
+import com.google.netpcapanalysis.interfaces.dao.MaliciousIPDao;
+import com.google.netpcapanalysis.models.PCAPdata;
+import com.google.netpcapanalysis.models.MaliciousPacket;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import com.google.netpcapanalysis.models.Flagged;
+
+import com.google.netpcapanalysis.dao.KeystoreDaoImpl;
+import com.google.netpcapanalysis.interfaces.dao.KeystoreDao;
+
+public class  MaliciousIPDaoImpl implements MaliciousIPDao{
+
+      private KeystoreDao keystoreDao = new KeystoreDaoImpl();
+    private final String AUTH0_API_KEY = keystoreDao.getKeystore().getAuth0APIKey();
+
+    private static final String FLAGGED_FALSE = "Resource not found";
+    private static final String FLAGGED_TRUE = "200: OK";
+
+    private PCAPDao ipCache = new PCAPDaoImpl();
+
+    public MaliciousIPDaoImpl(){
+    }
+    public PCAPdata isMalicious(PCAPdata data)
+    {
+        HttpResponse<String> result;
+
+        if(ipCache.searchMaliciousDB(data.destination) == true)
```

Comment:
You might want to change this logic a bit. The problem is if something is not malicious, you will still query the External Service even though you might have already seen it. So you will also need to cache even valid ones.